core/src/main/scala/kafka/coordinator/group/GroupMetadataManager.scala

```diff
@@ -311,37 +312,40 @@ class GroupMetadataManager(brokerId: Int,
 
           responseCallback(responseError)
         }
-        appendForGroup(group, groupMetadataRecords, putCacheCallback)
-
+        appendForGroup(group, groupMetadataRecords, putCacheCallback, completeDelayedRequests)
       case None =>
         responseCallback(Errors.NOT_COORDINATOR)
-        None
+        Map.empty
     }
   }
 
   private def appendForGroup(group: GroupMetadata,
                              records: Map[TopicPartition, MemoryRecords],
-                             callback: Map[TopicPartition, PartitionResponse] => Unit): Unit = {
+                             callback: Map[TopicPartition, PartitionResponse] => Unit,
+                             completeDelayedRequests: Boolean): Map[TopicPartition, Boolean] = {
     // call replica manager to append the group message
     replicaManager.appendRecords(
       timeout = config.offsetCommitTimeoutMs.toLong,
       requiredAcks = config.offsetCommitRequiredAcks,
       internalTopicsAllowed = true,
       origin = AppendOrigin.Coordinator,
+      completeDelayedRequests = completeDelayedRequests,
       entriesPerPartition = records,
       delayedProduceLock = Some(group.lock),
       responseCallback = callback)
   }
 
   /**
    * Store offsets by appending it to the replicated log and then inserting to cache
+   * @return the topic partitions having new records
```

Comment:
A map containing the topic partitions having new records and a flag indicating whether the HWM has been incremented.